### PR TITLE
ISSUES-3985 Support MySQL engine reading Enums type

### PR DESCRIPTION
**Pull request by bohutang**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Support MySQL engine reading Enums type #3985

### CODE REVIEW

File: tests/integration/test_storage_mysql/test.py [hunk 1]
```diff
@@ -114,6 +115,19 @@ def test_table_function(started_cluster):
...
 
 
+def test_enum_type(started_cluster):
+    table_name = 'test_enum_type'
+    conn = get_mysql_conn()
+    create_mysql_table(conn, table_name)
+    node1.query('''
+CREATE TABLE {}(id UInt32, name String, age UInt32, money UInt32, source Enum8('IP' = 1, 'URL' = 2)) ENGINE = MySQL('mysql1:3306', 'clickhouse', '{}', 'root', 'clickhouse', 1);
+'''.format(table_name, table_name))
+    node1.query("INSERT INTO {}(id, name, age, money) VALUES (1,'name',0, 0)".format(table_name))
```
Comment (alexey-milovidov) [inline tests/integration/test_storage_mysql/test.py:125]: Is it possible to INSERT `source` (enum column)?

Comment (BohuTANG) [reply tests/integration/test_storage_mysql/test.py:125]: let me have a go

**Review by alexey-milovidov — APPROVED** [review]
Ok.

BTW, we can also convert Enum to Enum, it will be more efficient. But it's not needed in this PR.

**Comment by BohuTANG** [comment]
Yes.
That's the universal solution for all the external engines(mongodb, cassandra etc) if they have Enum types.
Since mysqlxx(other storage driver) returns real enum string result not the enum type value, so need more works to do with it(need convert the string result to the Enum8 index)

**Comment by BohuTANG** [comment]
BTW, thanks for giving 'can be tested' free pass.

**Comment by alexey-milovidov** [comment]
> Integration tests

Broken in #11903

**Pull request merged**